shaders/circle.vertex.glsl

```diff
@@ -22,10 +31,32 @@ varying vec2 v_extrude;
 varying lowp vec4 v_color;
 varying lowp float v_antialiasblur;
 
+float evaluate_zoom_function_1(const vec4 values, const float t) {
+    if (t < 1.0) {
+        return mix(values[0], values[1], t);
+    } else if (t < 2.0) {
+        return mix(values[1], values[2], t - 1.0);
+    } else {
+        return mix(values[2], values[3], t - 2.0);
+    }
+}
+
+vec4 evaluate_zoom_function_4(const vec4 value0, const vec4 value1, const vec4 value2, const vec4 value3, const float t) {
+    if (t < 1.0) {
+        return mix(value0, value1, t);
+    } else if (t < 2.0) {
+        return mix(value1, value2, t - 1.0);
+    } else {
+        return mix(value2, value3, t - 2.0);
+    }
+}
+
```

Comment:
If we want to bring this to all other shaders, it will start growing very complex and repetitive... Perhaps we should consider a GLSL build system like https://github.com/stackgl/glslify